traitsui/testing/tester/wx/helpers.py

```diff
@@ -0,0 +1,62 @@
+#  Copyright (c) 2005-2020, Enthought, Inc.
+#  All rights reserved.
+#
+#  This software is provided without warranty under the terms of the BSD
+#  license included in LICENSE.txt and may be redistributed only
+#  under the conditions described in the aforementioned license.  The license
+#  is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+#  Thanks for using Enthought open source!
+#
+
+import wx
+
+
+def mouse_click_button(wrapper, interaction):
+    """ Performs a mouce click on a wx button.
+
+    Parameters
+    ----------
+    wrapper : UIWrapper
+        The wrapper object wrapping the wx button.
+    interaction : instance of traitsui.testing.tester.command.MouseClick
+        interaction is unused here, but it is included so that the function
+        matches the expected format for a handler.  Note this handler is
+        intended to be used with an interaction_class of a MouseClick.
+    """
+    control = wrapper.target
+    if not control.IsEnabled():
+        return
+    wx.MilliSleep(wrapper.delay)
+    click_event = wx.CommandEvent(
+        wx.wxEVT_COMMAND_BUTTON_CLICKED, control.GetId()
+    )
+    control.ProcessEvent(click_event)
+
+
+def mouse_click_ImageButton(wrapper, interaction):
+    """ Performs a mouce click on an pyface.ui.wx.ImageButton object.
+
+    Parameters
+    ----------
+    wrapper : UIWrapper
+        The wrapper object wrapping the ImageButton.
+    interaction : instance of traitsui.testing.tester.command.MouseClick
+        interaction is unused here, but it is included so that the function
+        matches the expected format for a handler.  Note this handler is
+        intended to be used with an interaction_class of a MouseClick.
+    """
+
+    control = wrapper.target
+    if not control.IsEnabled():
+        return
+    wx.MilliSleep(wrapper.delay)
+
+    left_down_event = wx.MouseEvent(
+        wx.wxEVT_LEFT_DOWN
+    )
+    left_up_event = wx.MouseEvent(
+        wx.wxEVT_LEFT_UP
+    )
+    control.ProcessEvent(left_down_event)
+    control.ProcessEvent(left_up_event)
```

Comment:
This works as intended, but Im not sure if this is the best way to emulate a Custom Style Button being clicked.  When using Custom Editor, the CustomEditor creates an `_control` attribute that is an ImageButton instance, and then the control of that is assigned as the editors control.  That control is a wx.Window (see the added handlers in wx Default Registry).  Since it is a window, normally, checks are performed to ensure the location of the click is correct 
```wdx, wdy = control.GetClientSize().Get()
        x, y = event.GetX(), event.GetY()
        control.Refresh()
        if (0 <= x < wdx) and (0 <= y < wdy):
            if self.style == "radio":
                self.selected = True
            elif self.style == "checkbox":
                self.selected = not self.selected
            else:
                self.clicked = True
```
(from https://github.com/enthought/pyface/blob/master/pyface/ui/wx/image_button.py)

@kitchoi should I try to update this to specify a location to click?  

Sldo the name specifies ImageButton as that was its intended use, but in reality the control is just a wx.Window, and no specifics to ImageButton are used.  Unsure if I should change the name and leave it only depending on wx.Window, or if I should rewrite the function to be more ImageButton specific.

Reply:
> This works as intended, but Im not sure if this is the best way to emulate a Custom Style Button being clicked.

This looks like a decent solution to me.

> should I try to update this to specify a location to click?

I did not notice that wx custom button editor has these many options.
Let's keep this simple and limit ourselves to testing the `style="button"` behaviour for now.

> Sldo the name specifies ImageButton as that was its intended use, but in reality the control is just a wx.Window, and no specifics to ImageButton are used. Unsure if I should change the name and leave it only depending on wx.Window, or if I should rewrite the function to be more ImageButton specific.

I am with you on this dilemma. I would guess that this function will evolve to become more ImageButton specific. Perhaps we should move this function to the module specific to wx ButtonEditor implementations and keep the name to be specific to ImageButton. Then we can generalize later when it becomes clearer this function can be generalized. Given this function will be called directly by name, it will be easy to track this down and move this out to a more shareable module later.